tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -1467,6 +1467,59 @@ tensorflow::Status ConvertPlugin(Converter& ctx,
   return tensorflow::Status::OK();
 }
 
+tensorflow::Status ConvertTranspose(
+    Converter& ctx, const tensorflow::NodeDef& node_def,
+    const std::vector<TRT_TensorOrWeights>& inputs,
+    std::vector<TRT_TensorOrWeights>* outputs) {
+  if (inputs.size() != 2 || !inputs.at(0).is_tensor() ||
+      !inputs.at(1).is_weights()) {
+    return tensorflow::errors::InvalidArgument(
+        "Input expects tensor and weights, at", node_def.name());
+  }
+  nvinfer1::ITensor* input_tensor = const_cast<nvinfer1::ITensor*>(
+      inputs.at(0).tensor());
+
+  TRT_ShapedWeights weights = inputs.at(1).weights();
+  const int* weights_ptr = static_cast<int*>(const_cast<void*>(
+      weights.GetValues()));
+  std::vector<int> perm(weights.count());
+  for (int i = 0; i < weights.count(); i++) {
+    perm[i] = weights_ptr[i];
+  }
+
+  nvinfer1::ITensor* output_tensor = ctx.TransposeTensor(input_tensor, perm);
+  outputs->push_back(TRT_TensorOrWeights(output_tensor));
+  return tensorflow::Status::OK();
+}
+
+tensorflow::Status ConvertReshape(
+    Converter& ctx, const tensorflow::NodeDef& node_def,
+    const std::vector<TRT_TensorOrWeights>& inputs,
+    std::vector<TRT_TensorOrWeights>* outputs) {
+  if (inputs.size() != 2 || !inputs.at(1).is_weights()) {
+    return tensorflow::errors::InvalidArgument(
+        "Input expects weights for shape, at", node_def.name());
+  }
+
+  TRT_ShapedWeights weights = inputs.at(1).weights();
+  const int* weights_ptr = static_cast<int*>(const_cast<void*>(
+      weights.GetValues()));
+  nvinfer1::Dims new_shape;
+  // Ignore first (batch) dimension because TRT abstracts batch away
```

Comment:
We need to return an error if the first dim doesn't match the max batch size of the engine.

Reply:
We also need to return an error if `weights.count() == 0` which means reshaping to scalar according to the [API doc](https://www.tensorflow.org/api_docs/python/tf/manip/reshape).

Reply:
And please add corresponding test cases.

Reply:
> We need to return an error if the first dim doesn't match the max batch size of the engine.

Would this work is the first dim == -1?

Would this work? new_shape and input_shape include the batch dim.
`
if new_shape[0] == -1 and product(new_shape[1:]) != product(original_shape[1:]):
  # Good
elif new_shape[0] == max_batch_size:
  # Good
else:
  # Bad
`

Reply:
Good catch, that should work, thanks!